src/AST-Core/RBMethodNode.class.st

```diff
@@ -683,6 +682,14 @@ RBMethodNode >> selectorParts [
 	^ self keywords collect: [:each | each asSymbol]
 ]
 
+{ #category : #'accessing compiled method' }
+RBMethodNode >> semanticScope: aSemanticScope [	
```

Comment:
I would prefer to stop crippling compiler stuff in the AST.

Reply:
I don't think I introduced some new dependency. It's a refactoring of what we have now regarding the method vs doIt compilation and semantics analysis. And being related to the semantics analysis it is packaged under compiler package as all other code for semantics analysis.

I agree that the current dependencies are not good. And this PR is my attempt to provide a foundation to fix this situation. 
The current code paths contain many kind of isDoit checks and they are quite difficult to follow. RBSemanticScope is introduced to remove such checks and simplify the logic. But it is still very initial version (while it brings a nice feature for temps access from doIts).

Reply:
I don't think I introduced some new dependency. It's a refactoring of what we have now regarding the method vs doIt compilation and semantics analysis. And being related to the semantics analysis it is packaged under compiler package as all other code for semantics analysis.

I agree that the current dependencies are not good. And this PR is my attempt to provide a foundation to fix this situation.
The current code paths contain many kind of isDoit checks and they are quite difficult to follow. RBSemanticScope is introduced to remove such checks and simplify the logic. But it is still very initial version (while it brings a nice feature for temps access from doIts).